src/components/ProgressRow/ProgressRow.jsx

```diff
@@ -0,0 +1,108 @@
+import React, {useEffect, useState} from "react";
+import './ProgressRow.css';
+
+const ProgressRow = ({taskList}) => {
+
+    const [boards, setBoards] = useState([]);
+
+    useEffect(() => {
+        setBoards([
+            {id: 1, title: "TO DO", items: taskList},
+            {id: 2, title: "READY FOR DEV", items: []},
+            {id: 3, title: "IN PROGRESS", items: []},
+            {id: 4, title: "DONE", items: []},
+        ]);
+    }, [taskList]);
+
+    const [currentBoard, setCurrentBoard] = useState(null);
+    const [currentItem, setCurrentItem] = useState(null);
+
+    function dragOverHandler(e) {
+        e.preventDefault()
+        if (e.target.className == 'box-container') {
+            e.target.style.boxShadow = '0 4px 3px gray'
+        }
+    }
+
+    function dragLeaveHandler(e) {
+        e.target.style.boxShadow = 'none'
+    }
+
+    function dragStartHandler(e, board, item) {
+        setCurrentBoard(board)
+        setCurrentItem(item)
+    }
+
+    function dragEndHandler(e) {
+        e.target.style.boxShadow = 'none'
+    }
+
+    function dropHandler(e, board, item) {
+        e.preventDefault();
+        const updatedBoards = boards.map((b) => {
+            if (b.id === board.id) {
+                const currentIndex = currentBoard.items.indexOf(currentItem);
+                currentBoard.items.splice(currentIndex, 1);
+                const dropIndex = b.items.indexOf(item);
+                b.items.splice(dropIndex + 1, 0, currentItem);
+                return { ...b, items: [...b.items] }; // Создание новой копии доски с обновленными элементами
+            }
+            if (b.id === currentBoard.id) {
+                return { ...currentBoard, items: [...currentBoard.items] }; // Создание новой копии доски с обновленными элементами
+            }
+            return b;
+        });
+
+        setBoards(updatedBoards);
+    }
+
+    function dropCardHandler(e, board) {
```

Comment:
useCallback with needed dependencies in the array of dependencies (such as currentBoard)